### ISSUE TITLE
The order of bands for multi-temporal data

### ISSUE THREAD

**Issue by FatemehEsfahani** [IBM/terratorch#777]
I have a dataset with two time steps. I stacked the input bands in the following order:
(**B_T0, G_T0, R_T0, NIR_0, SWIR1_T0, SWIR2_T0, B_T1, G_T1, R_T1, NIR_1, SWIR1_T1, SWIR2_T1**), where T0 and T1 correspond to the first and second time steps, respectively. 
I trained a model using this input format, but the results on test data were not satisfactory.
I am wondering if the band ordering that I used is correct? Unfortunately, I could not find any documentation about how to structure multi-temporal inputs.
Some researchers use the format instead: (**B_T0, B_T1, G_T0, G_T1, R_T0,  R_T1, NIR_0, NIR_1, SWIR1_T0,  SWIR1_T1, SWIR2_T0, SWIR2_T1**) to stack their temporal data, but I am not sure which approach is correct. Can someone help in this regards,
Thanks

**Comment by Joao-L-S-Almeida** [comment]
Hi, @FatemehEsfahani . I understand you are restoring a backbone, let us say, for Prithvi, and redefining the embedding layer (PatchEmbed) to support a different set of input bands, right ? In this context, I don't see the band ordering as being necessarily restricted, since part of the network is adapted to your new setup, but I think something similar to layout 1 would be more reasonable for an update. 
Are you leaving the rest of the backbone be update (freeze_backbone=False) ? 
Do you have any opinion, @paolofraccaro, @blumenstiel, @daniszw ?

**Comment by romeokienzler** [comment]
@FatemehEsfahani mind sharing the config with us?

**Comment by FatemehEsfahani** [comment]
Hi @romeokienzler @Joao-L-S-Almeida 
Actually I just want to do image segmentation on my input data using the config files that have been provided. My input data has two time frames.
Here are the parameters that I used: 


`EPOCHS = 100`
`SEED=0`
`OUT_DIR ="Experiment_v1_Prithvi `
`pl.seed_everything(SEED)`
`logger = TensorBoardLogger(
    save_dir=OUT_DIR,
    name="test",
)`
`callbacks = [
    RichProgressBar(),
    LearningRateMonitor(logging_interval="epoch"),
    EarlyStopping(monitor="val/loss", patience=40) 
]`
`trainer = pl.Trainer(
    accelerator="auto",
    strategy="auto",
    devices="auto",
    num_nodes=1,
    precision="16-mixed",
    logger=logger,
    callbacks=callbacks,
    max_epochs=EPOCHS,
    check_val_every_n_epoch=1,
    log_every_n_steps=5,
    enable_checkpointing=True,
    default_root_dir = OUT_DIR
)
`
`import os`
`os.environ['CURL_CA_BUNDLE'] = ''
`
`NUM_FRAMES = 2`
`LR = 1.5e-5`
`WEIGHT_DECAY = 0.05`

`backbone_args = dict(
    backbone_pretrained=True,
    backbone="prithvi_eo_v2_300", 
    backbone_pretrain_img_size=128,
    backbone_in_channels=6,
    backbone_bands=dataset_bands,
    backbone_num_frames=NUM_FRAMES,
)`

`decoder_args = dict(
    decoder="FCNDecoder",
    decoder_channels=512,
)`

`model_args = dict(
    **backbone_args,
    **decoder_args,
    num_classes=NUM_CLASSES,
    head_dropout=0.1,
    head_channel_list = [128, 64],
    rescale=False
)`

`model = SemanticSegmentationTask(
    model_args=model_args,
    loss="ce",
    class_names=["Background", "Deforestation"],
    class_weights=[0.3,1.5],
    freeze_backbone=False,
    freeze_decoder=False,
    model_factory="EncoderDecoderFactory",
    lr=LR,
    optimizer="AdamW",
    optimizer_hparams=dict(weight_decay=WEIGHT_DECAY),
    scheduler="ReduceLROnPlateau",
)`

> Hi, [@FatemehEsfahani](https://github.com/FatemehEsfahani) . I understand you are restoring a backbone, let us say, for Prithvi, and redefining the embedding layer (PatchEmbed) to support a different set of input bands, right ? In this context, I don't see the band ordering as being necessarily restricted, since part of the network is adapted to your new setup, but I think something similar to layout 1 would be more reasonable for an update. Are you leaving the rest of the backbone be update (freeze_backbone=False) ? Do you have any opinion, [@paolofraccaro](https://github.com/paolofraccaro), [@blumenstiel](https://github.com/blumenstiel), [@daniszw](https://github.com/daniszw) ?

**Comment by romeokienzler** [comment]
@FatemehEsfahani closing due to inactivute, feel free to reopen if needed
